core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -153,24 +153,6 @@ IndicesQueriesRegistry indicesQueriesRegistry() {
         return indicesQueriesRegistry;
     }
 
-    //norelease this needs to go away
-    public ParsedQuery parse(QueryBuilder queryBuilder) {
```

Comment:
just call toQuery on it, we can add back something that makes sense when we need it, but this one was simply misleading, it went from QueryBuilder back to json, back to query builder through parser, and then finally to lucene query. Only the last step is actually needed, which is calling toQuery.